src/App.vue

```diff
@@ -1,96 +1,106 @@
       isRecording: false,
-      hasMediaDeviceCaps: false,
-      startedRecording: null,
-      stoppedRecording: null,
-      mediaRecorder: null,
-      isFirefox: navigator.userAgent.match(/Firefox/),
-      isAndroid: navigator.userAgent.match(/Android/),
-      isEdge:
-        navigator.userAgent.match(/Edge/) !== -1 &&
-        (!!navigator.msSaveOrOpenBlob || !!navigator.msSaveBlob),
-      isSafari: !!navigator.userAgent.match(/^((?!chrome|android).)*safari/i),
-      isWin:
-        navigator.platform &&
-        navigator.platform
-          .toString()
-          .toLowerCase()
-          .indexOf("win") === -1
+      recordingStartedAt: null,
+      recordingEndedAt: null,
+      recorder: null,
+      recorderOptions: this.options || DEFAULT_OPTIONS
     };
   },
-  mounted() {
-    this.grantPermissions();
+  async mounted() {
+    this.isInitiated &&
+      (await this.initiatePlayer().catch((err) => {
+        this.isInitiated = false;
+        this.causeError(err);
+      }));
   },
   methods: {
-    async grantPermissions() {
-      let t = this;
-      try {
-        await navigator.mediaDevices.getUserMedia({ audio: true });
-        this.hasMediaDeviceCaps = true;
-      } catch (err) {
-        // could be used to show a hint that the device/browser does not support WebRTC.
-        // should potentially never be the case but better be safe
-        this.hasMediaDeviceCaps = false;
-      }
+    causeError(error) {
+      console.error("@loquiry/vue-record-audio error");
+      this.$emit("error", error);
+      throw error;
     },
-    async record() {
-      if (!this.isRecording) {
-        this.isRecording = true;
-        const stream = await navigator.mediaDevices.getUserMedia({
-          audio: this.isEdge
-            ? true
-            : {
-                echoCancellation: false
-              }
-        });
-
-        var options = {
-          type: "audio",
-          numberOfAudioChannels: this.isEdge ? 1 : 2,
-          checkForInactiveTracks: true,
-          bufferSize: 16384,
-          // disble webRTC logs
-          disableLogs: true
-        };
-
-        if (this.isSafari || this.isEdge || this.isFirefox) {
-          options.recorderType = RecordRTC.StereoAudioRecorder;
-        }
+    async initiatePlayer() {
+      let t = this;
 
-        if (this.isSafari) {
-          options.sampleRate = 44100;
-          options.bufferSize = 4096;
-        }
+      const audioContext = new (window.AudioContext ||
+        window.webkitAudioContext)();
 
-        this.mediaRecorder = new RecordRTC(stream.clone(), options);
-        this.mediaRecorder.startRecording();
-        this.startedRecording = performance.now();
-      } else {
-        this.isRecording = false;
-        this.stoppedRecording = performance.now();
-        if (this.mediaRecorder) {
-          await this.mediaRecorder.stopRecording(this.handleDataAvailable);
-        }
-      }
+      await navigator.mediaDevices
```

Comment:
there should be no need for async/await, in none of the occasions

Reply:
yeah I'm just learning about async/await and may have misused it in some cases :D thanks!